File: Core/src/main/java/dev/lynith/core/utils/NanoVGManager.java
package dev.lynith.core.utils;

import org.lwjgl.nanovg.NanoVG;
import org.lwjgl.nanovg.NanoVGGL3;

public class NanoVGManager {

    public static long createContext() {
        try {
            final long nvg = NanoVGGL3.nvgCreate(NanoVGGL3.NVG_ANTIALIAS | NanoVGGL3.NVG_STENCIL_STROKES);

            if (nvg == 0)
                throw new RuntimeException("Could not initialize NanoVG");

            NanoVG.nvgShapeAntiAlias(nvg, true);

            return nvg;
        } catch (Exception e) {
            e.printStackTrace();
            System.exit(1);
        }

        // Does this ever get reached?
        return -1;
    }

}

File: Core/src/main/kotlin/dev/lynith/core/ui/nvg/NanoVGManager.kt
package dev.lynith.core.ui.nvg

import org.lwjgl.nanovg.NanoVG
import org.lwjgl.nanovg.NanoVGGL2
import org.lwjgl.nanovg.NanoVGGL3

object NanoVGManager {
    fun createContext(): Long {
        try {
            val nvg = NanoVGGL3.nvgCreate(NanoVGGL3.NVG_ANTIALIAS or NanoVGGL3.NVG_STENCIL_STROKES)
            if (nvg == 0L) throw RuntimeException("Could not initialize NanoVG")

            NanoVG.nvgShapeAntiAlias(nvg, true)

            return nvg
        } catch (e: Exception) {
            e.printStackTrace()
            System.exit(1)
        }

        // Does this ever get reached?
        return -1
    }
}
